stdlib/public/CMakeLists.txt

```diff
@@ -19,6 +19,20 @@ if(SWIFT_STDLIB_SIL_DEBUGGING)
   list(APPEND SWIFT_RUNTIME_SWIFT_COMPILE_FLAGS "-Xfrontend" "-sil-based-debuginfo")
 endif()
 
+option(SWIFT_RUNTIME_PART_OF_DARWIN_SHARED_CACHE
+       "On Darwin platforms, link the standard library in a way that
+       allows inclusion in the dyld shared cache"
+       FALSE)
+
+if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+    set(linker_detected "$<BOOL:${LLVM_LINKER_DETECTED}>")
```

Comment:
For `LLVM_LINKER_DETECTED` to catch properly all the supported Apple linkers, we would need to amend the linker check for macOS -- I have a [LLVM PR for this](https://github.com/llvm/llvm-project/pull/77806)